src/text_formatter/text_formatter.ts

```diff
@@ -0,0 +1,158 @@
+ * @return a string-array where each element is one word
+ */
+export function sentenceIntoWordList(input: String) {
+	const separator = /[ ,]+/; // List all the values where the content of a sentence will get splitted: whitespace[ ] and comma[,]
+	const splittedSentenceIntoListOfWords = input.split(separator);
+	const temp = [];
+	for (let index = 0; index < splittedSentenceIntoListOfWords.length; index++) {
+		// remove all elements(senteces) which are a empty String after splitting
+		if (splittedSentenceIntoListOfWords[index].trim() == "") {
+			splittedSentenceIntoListOfWords.splice(index, 1);
+		} else {
+			temp.push(splittedSentenceIntoListOfWords[index].trim());
+		}
+	}
+	const ListOfWords = temp;
+	return ListOfWords;
+
+}
+
+/**
+* this function is the formatter for expressions; it calls "detectExpressionElements", "removeWhiteSpaces" and "replaceInputCaractersToReadablePrologCharacter";
+* ---------------------------------------
+* @param a input string of an expression where all operator has been formatted
+* @return a string array where each element is a element of the expression
+*/
+export function expressionFormatter(input: string) {
+	const detectedExpression = detectExpressionElements(input);
+	const separators = /(bracketLeft|bracketRight|equivalence|implication|negation|conjunction|disjunction)+/g;
+	const splittedExpression = detectedExpression.split(separators);
+	const temp = [];
+	for (let index = 0; index < splittedExpression.length; index++) {
+		let tempString = removeWhiteSpaces(splittedExpression[index]);
+		if (tempString !== "") {
+			tempString = replaceInputCaractersToReadablePrologCharacter(tempString);
+			temp.push(tempString);
+		}
+	}
+	const finalExpressionArray = temp;
+	return finalExpressionArray;
+}
+
+/**
+* remove every type of whitespace inside the expression
+* ---------------------------------------
+* @param a string
+* @return same string without any whitespaces left
+*/
+export function removeWhiteSpaces(input: string) {
+	const output = input.replace(/\s+/g, ""); // delete all white spaces
+	return output;
+}
+
+/**
+* replace expression elements into readable prolog Code
+* ---------------------------------------
+* @param 
+* @return
+*/
+export function replaceInputCaractersToReadablePrologCharacter(input: string) {
+	input = input.replace(/(bracketLeft)/g, "\[");
+	input = input.replace(/(bracketRight)/g, "\]");
+	input = input.replace(/(equivalence)/g, "<->");
+	input = input.replace(/(implication)/g, "->");
+	input = input.replace(/(negation)/g, "neg");
+	input = input.replace(/(conjunction)/g, "and");
+	input = input.replace(/(disjunction)/g, "or");
+	return input;
+}
+
+/**
+* check the expression if the amount of opened and closed brackets is correct; it can detect user mistakes; RETURN true if the user made a mistake;
+* ---------------------------------------
+* @param
+* @return
+*/
+export function checkAmountOfBrackets(formattedExpression: string[]) {
```

Comment:
But what if the user does something like `)test(`? Shouldn't that also complain about incorrect brackets? I've actually seen the problem of detecting correct bracketing being used as a developer interview question :D (it's not hard to implement)